Fix mypy errors for github module

```
theHarvester/discovery/githubcode.py:59: error: Incompatible types in assignment (expression has type "List[Any]", variable has type "str")  [assignment]
theHarvester/discovery/githubcode.py:61: error: Incompatible return value type (got "Union[str, int]", expected "Optional[int]")  [return-value]
```

--- a/theHarvester/discovery/githubcode.py
+++ b/theHarvester/discovery/githubcode.py
@@ -56,8 +56,8 @@ class SearchGithubCode:
         if page_link:
             parsed = urlparse.urlparse(page_link.get("url"))
             params = urlparse.parse_qs(parsed.query)
-            page = params.get('page') or [None]
-            page_number = page[0] and int(page[0])
+            pages = params.get('page', [None])
+            page_number = pages[0] and int(pages[0])
             return page_number
         else:
             return None

